src/olympia/scanners/views.py

```diff
@@ -4,27 +4,50 @@
 from django.db.models import CharField, Value
 from django.db.transaction import non_atomic_requests
 from django.http import Http404
+from rest_framework.exceptions import ParseError
 from rest_framework.generics import ListAPIView
 
 from olympia import amo
-from olympia.constants.scanners import TRUE_POSITIVE, LABEL_BAD, LABEL_GOOD
+from olympia.constants.scanners import (
+    LABEL_BAD,
+    LABEL_GOOD,
+    SCANNERS,
+    TRUE_POSITIVE,
+)
 from olympia.api.permissions import GroupPermission
 
 from .models import ScannerResult
 from .serializers import ScannerResultSerializer
 
 
-class ScannerResultViewSet(ListAPIView):
+class ScannerResultView(ListAPIView):
     permission_classes = [
         GroupPermission(amo.permissions.ADMIN_SCANNERS_RESULTS_VIEW)
     ]
 
     serializer_class = ScannerResultSerializer
 
     def get_queryset(self):
+        label = self.request.query_params.get('label', None)
+        scanner = next(
```

Comment:
Nit: This feels like a kind of convoluted and overly complex way of doing this, but I suppose it composes it into a single statement. I would have imagined checking for `self.request.query_params.get('scanner')` first, as we don't need to check `SCANNERS` if the query param isn't there, and then, if `scanner` exists, validating that it also exists in `SCANNERS`. What you have here is more concise, but also seems harder to read/mentally parse.

Reply:
Part of the complexity is retrieving a key in a dict based on the value.